pkg/apis/dashboard/migration/conversion/conversion.go

```diff
@@ -0,0 +1,187 @@
+	setConversionAnno(&out.ObjectMeta, "V0_to_V1")
+	return nil
+}
+
+func Convert_V0_to_V2(in *dashboardV0.Dashboard, out *dashboardV2.Dashboard, scope conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Spec = dashboardV2.DashboardSpec{
+		Unstructured: in.Spec,
+	}
+	setConversionAnno(&out.ObjectMeta, "V0_to_V2")
+	return nil
+}
+
+func Convert_V1_to_Internal(in *dashboardV1.Dashboard, out *dashboard.Dashboard, scope conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Spec = dashboard.DashboardSpec(in.Spec)
+	setConversionAnno(&out.ObjectMeta, "V1_to_>>>>>>>")
+	return nil
+}
+
+func Convert_V1_to_V0(in *dashboardV1.Dashboard, out *dashboardV0.Dashboard, scope conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Spec = in.Spec.Unstructured
+	setConversionAnno(&out.ObjectMeta, "V1_to_V0")
+	return nil
+}
+
+func Convert_V1_to_V2(in *dashboardV1.Dashboard, out *dashboardV2.Dashboard, scope conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Spec = dashboardV2.DashboardSpec(in.Spec)
+	setConversionAnno(&out.ObjectMeta, "V1_to_V2")
+	return nil
+}
+
+func Convert_V2_to_Internal(in *dashboardV2.Dashboard, out *dashboard.Dashboard, scope conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Spec = dashboard.DashboardSpec(in.Spec)
+	setConversionAnno(&out.ObjectMeta, "V2_to_>>>>>>>")
+	return nil
+}
+
+func Convert_V2_to_V0(in *dashboardV2.Dashboard, out *dashboardV0.Dashboard, scope conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Spec = in.Spec.Unstructured
+	setConversionAnno(&out.ObjectMeta, "V2_to_V0")
+	return nil
+}
+
+func Convert_V2_to_V1(in *dashboardV2.Dashboard, out *dashboardV1.Dashboard, scope conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Spec = dashboardV1.DashboardSpec(in.Spec)
+	setConversionAnno(&out.ObjectMeta, "V2_to_V1")
+	return nil
+}
+
+func Convert_to_V0(in *dashboard.Dashboard, out *dashboardV0.Dashboard, scope conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Spec = in.Spec.Unstructured
+	setConversionAnno(&out.ObjectMeta, ">>>>>>__to_V0")
+	return nil
+}
+
+func Convert_to_V1(in *dashboard.Dashboard, out *dashboardV1.Dashboard, scope conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Spec = dashboardV1.DashboardSpec(in.Spec)
+	setConversionAnno(&out.ObjectMeta, ">>>>>>__to_V1")
+	return nil
+}
+
+func Convert_to_V2(in *dashboard.Dashboard, out *dashboardV2.Dashboard, scope conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Spec = dashboardV2.DashboardSpec(in.Spec)
+	setConversionAnno(&out.ObjectMeta, ">>>>>>__to_V2")
+	return nil
+}
+
+// Add an annotation tracking which conversions have been run
+func setConversionAnno(meta *v1.ObjectMeta, val string) {
+	anno := meta.Annotations
+	if anno == nil {
+		anno = make(map[string]string)
+		meta.Annotations = anno
+	}
+
+	for i := range 10 {
+		key := fmt.Sprintf("CONVERSION-%d", i)
+		if anno[key] != "" {
+			continue
+		}
+		anno[key] = val
```

Comment:
^^^ we could comment out this line if we do not want to merge adding the annotations

(but I think having the annotations is helpful until we can add real conversion info into the status)

Reply:
See https://github.com/grafana/grafana/pull/101531